File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.py
def solution(s):
    ans = ''
    flag = True
    for c in s:
        if flag:
            ans += c.upper()
        else: ans += c.lower()
        # 이전 문자가 공백이면 upper
        # c가 공백인지 확인(공백일 경우 upper해도 "")
        flag = c == " "
    return ans

File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.kt
class Solution {
    fun solution(s: String): String {
        var ans = ""
        var flag = true
        for (chr in s) {
            if (flag) {
                ans += chr.uppercase()
            } else {
                ans += chr.lowercase()
            }
            flag = chr == ' '
        }
        return ans
    }
}